### PR TITLE
feat: File Searching

### PR DESCRIPTION
**Pull request by cbolles**
* Basic file searching implementation
* Navigates user to contained folder

### CODE REVIEW

File: src/components/FileSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  // TODO: By introducing the concept of a file type can reduce some of this
+  // duplicated logic for determining if the object is a folder
+  const fileComponents = props.option.id.split('/');
+  const isFolder = fileComponents[fileComponents.length - 1] == '';
+
+
+  return (
+    <Grid container key={props.option.id}>
+      <Grid item xs={2}>
+        {isFolder ? <FolderIcon /> : <></>}
```
Comment (gregfrasco) [inline src/components/FileSearch.tsx:24]: nit
```suggestion
        {isFolder && <FolderIcon />}
```

Comment (gregfrasco) [inline src/components/FileSearch.tsx:24]: nit
```suggestion
        {isFolder && <FolderIcon />}
```

**Review by gregfrasco — APPROVED** [review]
Minor Changes

File: src/components/FileSearch.tsx (hunk 1 above)
Comment (cbolles) [reply src/components/FileSearch.tsx]: That's cleaner, thanks for the suggestion!

**Review by am5815 — APPROVED** [review]

**Pull request merged**